### PR TITLE
[T2A2][T09-A4]Cao Wei

### PR DESCRIPTION
**Pull request by LoneGunman001**

### CODE REVIEW

**Comment by LoneGunman001** [comment]
Ready for review

File: src/seedu/addressbook/AddressBook.java [hunk 1]
```diff
@@ -181,14 +181,14 @@
     /**
      * List of all persons in the address book.
      */
-    private static final ArrayList<String[]> ALL_PERSONS = new ArrayList<>();
+    private static final ArrayList<String[]> ALL_PERSONS = new ArrayList<String[]>();
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:184]: :+1:

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -181,14 +181,14 @@
...
 
     /**
      * Stores the most recent list of persons shown to the user as a result of a user command.
      * This is a subset of the full list. Deleting persons in the pull list does not delete
      * those persons from this list.
      */
-    private static ArrayList<String[]> latestPersonListingView = getAllPersonsInAddressBook(); // initial view is of all
+    private static ArrayList<String[]> latestPersonsListingView = getAllPersonsInAddressBook(); // initial view is of all
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:191]: good catch!

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -645,7 +645,7 @@ private static String getDisplayString(ArrayList<String[]> persons) {
             final int displayIndex = i + DISPLAYED_INDEX_OFFSET;
             messageAccumulator.append('\t')
                               .append(getIndexedPersonListElementMessage(displayIndex, person))
-                              .append(LS);
+                              .append(LINE_SEPARATOR);
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:648]: good! it is always good have a descriptive variable/constant/method name

File: src/seedu/addressbook/AddressBook.java [hunk 4]
```diff
@@ -1146,12 +1146,13 @@ private static String getUsageInfoForExitCommand() {
...
-     * @param trim  Parameter trim to be removed
+     * @param fullString  Parameter as a string
+     * @param prefix  Parameter trim to be removed
      * @return  string without the trim
      */
-    private static String removePrefixTrim(String s, String trim) {
-        return s.replace(trim, "");
+    private static String removePrefix(String fullString, String prefix) {
+        return fullString.replace(prefix, "");
+        //TODO
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java]: isn't this an unnecessary comment here?

File: src/seedu/addressbook/AddressBook.java [hunk 5]
```diff
@@ -1147,11 +1147,11 @@ private static String getUsageInfoForExitCommand() {
...
      * @param s  Parameter as a string
-     * @param sign  Parameter sign to be removed
-     * @return  string without the sign
+     * @param trim  Parameter trim to be removed
+     * @return  string without the trim
      */
-    private static String removePrefixSign(String s, String sign) {
-        return s.replace(sign, "");
+    private static String removePrefixTrim(String s, String trim) {
+        return s.replace(trim, "");
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java]: notice that the method still replaces all occurrences of the prefix pattern. please read the description carefully again

**Comment by okkhoy** [comment]
@LoneGunman001 while you have made a number of improvements to the code, notice that there is just one commit for the entire exercise. please ensure you commit after every change for better revision control.

please take a look at the comments, ack and close the pull request.

on an unrelated note, please add a photo to your profile to help us identify you better :-)

**Pull request closed without merging**